### DISCUSSION TITLE
How to change the directory with zoxide using a keybinding in Fish shell?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [ajeetdsouza/zoxide#791] [Q&A] [answered]
I have zoxide set up and working in my Fish shell. I can use z to jump to directories based on my history. However, I'd like to set up a custom keybinding to use zoxide more efficiently.

**Comment by ajeetdsouza** [comment]
You can read the documentation on Fish keybindings here: https://fishshell.com/docs/3.3/cmds/bind.html

Is there anything specific about zoxide you want to ask?

**Reply by limaceous-bushwhacker** [reply]
Yes, I'm not sure what to use in the command section. Currently, I'm using `bind <key> zi` but on hitting <key> I get the interactive menu but I can't seem to cd into the selected dir on pressing enter.

**Reply by ajeetdsouza** [reply, marked as answer]
The directory is likely getting changed, it's probably just the prompt that isn't being redrawn. Try:

```fish
bind <key> 'zi || commandline --function repaint'
```

**Reply by limaceous-bushwhacker** [reply]
Ty! That works as expected.

**Reply by mawkler** [reply]
Changing `&&` to `||` makes the prompt get properly re-rendered when pressing `ctrl-c` from inside Zoxide:

```sh
bind <key> zi repaint
```

**Reply by sondhg** [reply]
> Changing `&&` to `||` makes the prompt get properly re-rendered when pressing `ctrl-c` from inside Zoxide:
> 
> ```shell
> bind <key> zi repaint
> ```
This works for me:
```fish
function fish_user_key_bindings
    bind alt-r 'zi; commandline -f repaint'
    bind -M insert alt-r 'zi; commandline -f repaint'
end
```
